actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -9,14 +9,32 @@ module SubscriptionAdapter
     class PostgreSQL < Base # :nodoc:
       prepend ChannelPrefix
 
+      PayloadTooLargeError = Class.new(StandardError)
+
+      MAX_NOTIFY_SIZE = 7997 # documented as 8000 bytes, but there appears to be some overhead in transit
+      DEFAULT_LARGE_PAYLOADS_TABLE = "action_cable_large_payloads"
+      LARGE_PAYLOAD_PREFIX = "__large_payload:"
+
+      attr_reader :large_payloads_table
+
       def initialize(*)
         super
         @listener = nil
+        @large_payloads_table = @server.config.cable[:large_payloads_table] || DEFAULT_LARGE_PAYLOADS_TABLE
+        @insert_large_payload_query = "INSERT INTO #{@large_payloads_table} (payload) VALUES ($1) RETURNING id"
```

Comment:
Storing the query in an ivar is just a micro-optimisation to avoid String allocations for each invocation. Could be inlined if preferred. Same goes for the select query below.